configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java

```diff
@@ -31,53 +31,43 @@
     }
 
     @Override
-    public FileReference addFile(String relativePath) throws IOException {
-        Path path = Path.of(relativePath).normalize();
-        if (path.isAbsolute())
-            throw new IllegalArgumentException(relativePath + " is not relative");
-        File file = new File(applicationDir, relativePath);
-        Path relative = applicationDir.toPath().relativize(file.toPath()).normalize();
-        if (relative.isAbsolute() || relative.startsWith(".."))
-            throw new IllegalArgumentException(file + " is not a descendant of " + applicationDir);
-
-        return fileDirectory.addFile(file);
+    public FileReference addFile(Path path) throws IOException {
+        File file = new File(applicationDir, path.getRelative());
+        return addFile(file);
     }
 
-    @Override
-    public FileReference addFile(File file) throws IOException {
+    private FileReference addFile(File file) throws IOException {
         return fileDirectory.addFile(file);
     }
 
     @Override
-    public FileReference addUri(String uri, String relativePath) {
-        File file = download(uri, relativePath);
-        try {
-            return addFile(file);
-        } catch (IOException e) {
+    public FileReference addUri(String uri, Path path) {
+        throw new UnsupportedOperationException("URI type is not supported");
```

Comment:
But we already support this?

Reply:
No? The only thing I can find that calls this code is ONNX-models, but they also explicitly throw if the type is URI ...

Reply:
Ok!

Reply:
I was wrong 😭